src/gfx/reverse.cpp

```diff
@@ -171,8 +171,18 @@ void reverse() {
 			        options.nbPalettes);
 		}
 
-		if (options.palSpecType == Options::EXPLICIT && palettes != options.palSpec) {
-			warning("Colors in the palette file do not match those specified with `-c`!");
+		if (options.palSpecType == Options::EXPLICIT) {
+			for (auto &palette : palettes) {
+				for (auto &specPalette : options.palSpec) {
+					for (auto &color : palette) {
+						for (auto &specColor : specPalette) {
+							if (specColor && color != specColor.value()) {
+								warning("Colors in the palette file do not match those specified with `-c`!");
+							}
+						}
+					}
+				}
+			}
```

Comment:
```suggestion
		if (options.palSpecType == Options::EXPLICIT && palettes != options.palSpec) {
			warning("Colors in the palette file do not match those specified with `-c`!");
```

Now that both are the same type, can they be compared directly again?

Reply:
Apparently yes.